### PR TITLE
Fixed #4671 : Hints & Solution and Concept Card Dark Mode

### LINKED ISSUE #4671: Hint screen contrast should be improved
**Describe the bug**
The contrast of the text and background color on the hint screen prevented the learner from having the best experience.

**To Reproduce**
1. Go to "Home, "select a topic to start"
2. Click on "Place Values"
3. Start a lesson "Chapter 2"
4. To see error "Click on the hint icon"

**Expected behavior**
The contrast of the text and background color on the hint screen prevented the learner from having the best experience which shouldn't be. The texts should be more readable.

**Demonstration**
image

**Environment**
-Device Used: Android
-App version: 0.9-beta-3f935261e0

### PR DESCRIPTION
**Pull request by MohitGupta121**
Fixed #4671

## Explanation

**Mockups:**

Concept Card Default: 
https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/2cc59921-923b-43ef-bf11-cf7cc5f7d903/specs/

Concept Card Dark Mode: 
https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/2ee531cf-c358-4a7d-b9dd-b772b51c3d3c/specs/

Hint Summary Default: 
https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/99e9e4c3-6c5e-4c13-a9f2-6b92aee17392/specs/

Hint Summary Dark Mode:
https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d43c5d96-945a-4b89-9552-c4fb482ce2f2/specs/
https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/a70b76a7-49ae-47b6-b5fe-f148b71ee620/specs/

Solution Summary Default:
https://xd.adobe.com/view/3dca36c2-5115-419c-b25e-0f10526b077c-6899/screen/545c6f32-2fec-4a8b-bb53-cecfb8e52f31/specs/

Solution Summary Dark Mode:
https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/5362a633-73cb-4288-aa7d-8c69e4eef349/specs/
https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/a50b0252-f883-451d-9729-dde8d34e2cc5/specs/

Fixed Hints & Solution and Concept Card Dark Mode

## Essential Checklist

- [ ] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [ ] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

|**Default - Concept Card**|**Dark Mode - Concept Card**|
|:----------------------------------------:|:-----------------------------------------:|
|  |  |
|**Default - Hint Summary**|**Dark Mode - Hint Summary**|
|  |  |
|||
|  |  |
|**Default - Solution Summary**|**Dark Mode - Solution Summary**|
|  |  |
|||
|  |  |
|||
|  |  |

### CODE REVIEW

File: app/src/main/res/layout/hints_summary.xml [hunk 1]
```diff
@@ -109,6 +109,7 @@
             app:layout_constraintVertical_bias="0.0"
             app:rotationAnimationAngle="@{180f}"
             app:srcCompat="@drawable/ic_arrow_drop_down_black_24dp"
+            android:background="@color/color_def_transparent"
```
Comment (rt4914) [inline app/src/main/res/layout/hints_summary.xml]: We can only use values from `component_colors.xml` file in any layout file.

You will need to create a `color_palette.xml* resource dark/light both and then use that in `component_colors.xml` and finally use that in here. 

Have a look at this https://github.com/oppia/oppia-android/wiki/Dark-Mode

File: app/src/main/res/layout/solution_summary.xml [hunk 2]
```diff
@@ -108,6 +108,7 @@
             app:layout_constraintVertical_bias="0.0"
             app:rotationAnimationAngle="@{180f}"
             app:srcCompat="@drawable/ic_arrow_drop_down_black_24dp"
+            android:background="@color/color_def_transparent"
```
Comment (rt4914) [inline app/src/main/res/layout/solution_summary.xml]: We can only use values from `component_colors.xml` file in any layout file.

You will need to create a `color_palette.xml* resource dark/light both and then use that in `component_colors.xml` and finally use that in here. 

Have a look at this https://github.com/oppia/oppia-android/wiki/Dark-Mode

**Review by rt4914 — CHANGES_REQUESTED** [review]
@MohitGupta121 PTAL. Thanks.

File: app/src/main/res/layout/hints_summary.xml (hunk 1 above)
Comment (MohitGupta121) [reply app/src/main/res/layout/hints_summary.xml]: Okay sure

File: app/src/main/res/layout/solution_summary.xml (hunk 2 above)
Comment (MohitGupta121) [reply app/src/main/res/layout/solution_summary.xml]: Okay I'll do it.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Comment by MohitGupta121** [comment]
Thanks @rt4914,
Can you please assign me next part of this dark mode project in doc.

**Pull request merged**